openupgradelib/openupgrade.py

```diff
@@ -3261,17 +3261,43 @@ def chunked(records, single=True):
     """Memory and performance friendly method to iterate over a potentially
     large number of records. Yields either a whole chunk or a single record
     at the time. Don't nest calls to this method."""
-    if version_info[0] > 10:
-        invalidate = records.env.cache.invalidate
-    elif version_info[0] > 7:
-        invalidate = records.env.invalidate_all
+    # Warning on possibly untested versions where chunked might not work as expected
+    if version_info[0] > 17:  # unreleased version at this time
+        logger.warning(
+            f"chunked method might not work as expected in Odoo {version_info[0]}. "
+            "Please report any issue you may find and consider bumping this warning "
+            "to the next version otherwise."
+        )
+    # Before each iteration we must flush everything and clear the cache.
+    # This differs depending on the Odoo version.
+    if version_info[0] >= 16:
+
+        def reset():
+            records.env.invalidate_all(flush=True)
+
+    elif version_info[0] >= 13:
+
+        def reset():
+            records.env["base"].flush()
+            records.env.cache.invalidate()
+
+    elif version_info[0] >= 11:
+
+        def reset():
+            records.env.cache.invalidate()
+
+    elif version_info[0] >= 8:
+
+        def reset():
+            records.env.invalidate_all()
+
     else:
         raise Exception("Not supported Odoo version for this method.")
     size = core.models.PREFETCH_MAX
     model = records._name
     ids = records.with_context(prefetch_fields=False).ids
     for i in range(0, len(ids), size):
-        invalidate()
+        reset()
```

Comment:
I like having a name like `invalidate` as it reflects the aim more clearly and is aligned with the underlying Odoo methods it calls. Can you tell my why you changed it?

Reply:
My reasoning comes from the fact that we're not only invalidating the "read cache", we're also flushing the "to write cache"

Something like `flush_and_invalidate` seemed more accurate but it's long and explicit. Then I thought for each iteration a "reset" of the cache is done, hence the name.

In any case, I've extracted the invalidation/flushing to a separate method, as you suggested